### PR TITLE
/tg/ cherry-pick

### PR DESCRIPTION
**Pull request by ghost**
i cherry-picked stuff of the following categories from /tg/:

* bugfixes
* code refactors
* minor, kinda cool features that don't affect balance

no new balance changes or major features.

### CODE REVIEW

**Comment by ghost** [comment]
requires sql update
```INSERT INTO `schema_revision` (`major`, `minor`) VALUES (5, 2);```
```ALTER TABLE `player`
	ADD COLUMN `discord_id` BIGINT NULL DEFAULT NULL AFTER `flags`;```

**Comment by Carbonhell** [comment]
basically a smaller upstream merge

File: code/modules/antagonists/traitor/syndicate_contract.dm [hunk 1]
```diff
@@ -6,22 +6,19 @@
...
-	// 50/50 chance of getting at least one very high paying
-	// contract.
-	// High payout
-	if (prob(18))
-		contract.payout_bonus = rand(7,10)
-	else if (prob(45))
-		contract.payout_bonus = rand(1,3)
-	else // Medium payout
-		contract.payout_bonus = rand(2,6)
+	contract.find_target(null, blacklist)
```
Comment (Manly-Dwarf) [inline code/modules/antagonists/traitor/syndicate_contract.dm:14]: oh hey, will this fix the duplicate contracts bug?

**Comment by Manly-Dwarf** [comment]
I dunno if this is 100% necessary. It adds stacks, edits some spans, and that's about it.

**Comment by optimumtact** [comment]
image

**Comment by Manly-Dwarf** [comment]
ye that's my thought process exactly oranges
why is this necessary, didn't we just go through this whole discussion about how we don't want these any more

**Comment by optimumtact** [comment]
I dont' blame MRTY, he's your smartest coder by far so he definitely sees the value in the changes.

I don't expect the rest of you to understand though.

**Comment by Manly-Dwarf** [comment]
cleaner code is nice, there's always a value in that, but I'm pretty sure we cut off /tg/ for a reason. already movin to a cherrypick PR? barely been a month
the contractor stuff is neat though

**Comment by yoyobatty** [comment]
It took a lot of convincing and thinking for me to kill the mirror bot. Everyone seemed ambitious enough to want to make hippie great again but it seems those were just ideas that never left the runway.

**Comment by ghost** [comment]
The reason we killed the mirrorbot was due to /tg/'s poor balancing and crappy new features, right?

This bypasses that issue. By manually cherry-picking the commits, we never face that problem.

**Comment by ghost** [comment]
Imma redo this, i think i did it wrong, tho

**Pull request closed without merging**